community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/AbstractLuceneIndex.java

```diff
@@ -58,12 +60,14 @@
     protected final ReentrantLock partitionsLock = new ReentrantLock();
 
     protected final PartitionedIndexStorage indexStorage;
+    private final Supplier<IndexWriterConfig> writerConfigSupplier;
     private List<IndexPartition> partitions = new CopyOnWriteArrayList<>();
     private volatile boolean open;
 
-    public AbstractLuceneIndex( PartitionedIndexStorage indexStorage )
+    public AbstractLuceneIndex( PartitionedIndexStorage indexStorage, Supplier<IndexWriterConfig> writerConfigSupplier )
```

Comment:
Would propose to use Factory<IndexWriterConfig> instead since we need new instance always